src/client/src/system/SystemRenderAttack.cpp

```diff
@@ -25,7 +25,10 @@ void SystemRenderAttack::update(SystemStageRenderer& sr)
 
 		for (auto& target : targets)
 		{
-			irr::video::SColor color(255, 0, 0, 255 / (attack.beats_until_attack + 1));
+			irr::video::SColor color(255,
```

Comment:
Please perform this transformation in HSV space. I'd create the other helper functions for shifting saturation and value for better readability.

Reply:
That feels like a load of unnecessary conversions for a small amount of readability. I will add a comment but I don't think I'll be adding a conversion.